container/dangerzone.py

```diff
@@ -334,191 +269,97 @@ def pixels_to_pdf(self) -> int:
-                try:
-                    p = subprocess.run(
-                        args,
-                        stdout=subprocess.DEVNULL,
-                        stderr=subprocess.DEVNULL,
-                        timeout=60,
-                    )
-                except subprocess.TimeoutExpired:
-                    self.output(
-                        True,
-                        "Error converting pixels to PNG, convert timed out after 60 seconds",
-                        percentage,
-                    )
-                    return 1
-                if p.returncode != 0:
-                    self.output(
-                        True,
-                        f"Page {page}/{num_pages} conversion to PNG failed",
-                        percentage,
-                    )
-                    return 1
-
-                args = [
-                    "tesseract",
-                    png_filename,
-                    ocr_filename,
-                    "-l",
-                    os.environ.get("OCR_LANGUAGE"),  # type: ignore
-                    "--dpi",
-                    "70",
-                    "pdf",
-                ]
-                try:
-                    p = subprocess.run(
-                        args,
-                        stdout=subprocess.DEVNULL,
-                        stderr=subprocess.DEVNULL,
-                        timeout=60,
-                    )
-                except subprocess.TimeoutExpired:
-                    self.output(
-                        True,
-                        "Error converting PNG to searchable PDF, tesseract timed out after 60 seconds",
-                        percentage,
-                    )
-                    return 1
-                if p.returncode != 0:
-                    self.output(
-                        True,
-                        f"Page {page}/{num_pages} OCR failed",
-                        percentage,
-                    )
-                    return 1
-
-            else:
-                # Don't OCR
-                self.output(
-                    False,
-                    f"Converting page {page}/{num_pages} from pixels to PDF",
-                    percentage,
+            else:  # Don't OCR
+                self.update_progress(
+                    f"Converting page {page}/{num_pages} from pixels to PDF"
+                )
+                run_command(
+                    [
+                        "gm",
+                        "convert",
+                        "-size",
+                        f"{width}x{height}",
+                        "-depth",
+                        "8",
+                        f"rgb:{rgb_filename}",
+                        f"pdf:{pdf_filename}",
+                    ],
+                    error_message=f"Page {page}/{num_pages} conversion to PDF failed",
+                    timeout_message=f"Error converting RGB to PDF, convert timed out after {DEFAULT_TIMEOUT} seconds",
                 )
```

Comment:
We define the same `gm` command twice, regardless of whether we want OCR or not. Shouldn't we factor it out?

Reply:
Almost the same, but not exactly the same. [The first](https://github.com/freedomofpress/dangerzone/blob/097fe7d50d67ff9e609f14621892fe4c626f60a3/container/dangerzone.py#L272-L289) is from RGB to PNG, [the second](https://github.com/freedomofpress/dangerzone/blob/097fe7d50d67ff9e609f14621892fe4c626f60a3/container/dangerzone.py#L305-L322) is from RGB to PDF.

Reply:
Oh, you're right, thanks for pointing out. Still, I think we can do the factoring out (e.g., `gm_convert_to()`, and add the target file type and filename as arguments to that command).

Reply:
Sure, makes sense. We're good to go then!

@deeplow do you want to merge this, or should I go ahead?